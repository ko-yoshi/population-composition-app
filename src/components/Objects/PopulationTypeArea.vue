<template>
  <fieldset class="populations">
    <legend>グラフに表示する人口構成タイプを選択してください</legend>
    <template v-for="i in POPULATION_TYPE" :key="i.code">
      <PartsRadioBtn
        class="item"
        :label="i.name"
        name="radio"
        :checked="state.checkedPopulationType === i.code"
        @change="updatePopulationType(i.code)"
      />
    </template>
  </fieldset>
</template>
<script setup lang="ts">
  const { state } = useLineChartStore()
  const { updatePopulationType } = usePopulationComposition()
</script>
<style lang="scss" scoped>
  .populations {
    display: flex;
    width: fit-content;

    .item:not(:first-of-type) {
      margin-left: 0.5em;
    }
  }
</style>
