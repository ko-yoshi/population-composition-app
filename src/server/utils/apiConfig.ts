export const apiBaseUrl = () => {
  return useRuntimeConfig().public.apiBase
}

export const apiHeaders = () => {
  return { 'X-API-KEY': useRuntimeConfig().resasApiKey }
}

export type BaseResponse<T> = {
  message: string
  statusCode?: string
  description?: string
  result?: T
}
