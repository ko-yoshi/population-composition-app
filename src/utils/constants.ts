export const POPULATION_CODE = {
  TOTAL: 1,
  YOUNGER: 2,
  WORK: 3,
  ELDERLY: 4,
} as const

export const POPULATION_TYPE = {
  TOTAL: { code: POPULATION_CODE.TOTAL, name: '総人口' },
  YOUNGER: { code: POPULATION_CODE.YOUNGER, name: '年少人口' },
  WORK: { code: POPULATION_CODE.WORK, name: '生産年齢人口' },
  ELDERLY: { code: POPULATION_CODE.ELDERLY, name: '老年人口' },
} as const
