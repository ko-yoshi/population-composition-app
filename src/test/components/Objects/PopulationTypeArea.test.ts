import { mount } from '@vue/test-utils'
import { expect, test, describe, vi, beforeEach, afterEach } from 'vitest'
import PartsRadioBtn from '~/components/Parts/RadioBtn.vue'
import PopulationTypeArea from '~/components/Objects/PopulationTypeArea.vue'
import { POPULATION_TYPE } from '~/utils/constants'

vi.mock('~/composables/usePopulationComposition', () => ({
  usePopulationComposition: vi.fn(() => ({
    updatePopulationType: vi.fn(),
  })),
}))
vi.mock('~/composables/useLineChartStore', () => ({
  useLineChartStore: vi.fn(() => ({
    state: {
      value: { prefectures: [{ prefCode: 1, prefName: '北海道' }], checkedPopulationType: 1 },
    },
  })),
}))

describe('PopulationTypeArea', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = mount(PopulationTypeArea, {
      data() {
        return {
          POPULATION_TYPE,
        }
      },
    })
  })

  afterEach(() => {
    vi.clearAllMocks()
    wrapper.unmount()
  })

  test('人口構成オプションが表示されることを確認する', () => {
    const types = [
      { code: 1, name: '総人口' },
      { code: 2, name: '年少人口' },
      { code: 3, name: '生産年齢人口' },
      { code: 4, name: '老年人口' },
    ]

    const items = wrapper.findAllComponents(PartsRadioBtn)
    expect(items).toHaveLength(4)

    types.forEach((type, index) => {
      expect(items[index].props('label')).toBe(type.name)
      expect(items[index].props('name')).toBe('radio')
    })
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
