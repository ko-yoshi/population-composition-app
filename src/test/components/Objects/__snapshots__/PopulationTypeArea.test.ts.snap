// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopulationTypeArea > snapshot 1`] = `
"<fieldset data-v-b7f3fcd7=\\"\\" class=\\"populations\\">
  <legend data-v-b7f3fcd7=\\"\\">グラフに表示する人口構成タイプを選択してください</legend><label data-v-63b2aec9=\\"\\" data-v-b7f3fcd7=\\"\\" class=\\"radio-btn item\\"><input data-v-63b2aec9=\\"\\" type=\\"radio\\" name=\\"radio\\"> 総人口</label><label data-v-63b2aec9=\\"\\" data-v-b7f3fcd7=\\"\\" class=\\"radio-btn item\\"><input data-v-63b2aec9=\\"\\" type=\\"radio\\" name=\\"radio\\"> 年少人口</label><label data-v-63b2aec9=\\"\\" data-v-b7f3fcd7=\\"\\" class=\\"radio-btn item\\"><input data-v-63b2aec9=\\"\\" type=\\"radio\\" name=\\"radio\\"> 生産年齢人口</label><label data-v-63b2aec9=\\"\\" data-v-b7f3fcd7=\\"\\" class=\\"radio-btn item\\"><input data-v-63b2aec9=\\"\\" type=\\"radio\\" name=\\"radio\\"> 老年人口</label>
</fieldset>"
`;
